src/autopas/utils/SimilarityFunctions.h

```diff
@@ -28,86 +46,68 @@ template <class Container>
                                                             const std::array<double, 3> &startCorner,
                                                             const std::array<double, 3> &endCorner) {
-  unsigned int numberOfParticles = static_cast<unsigned int>(container.getNumberOfParticles());
-  autopas::AutoPas_MPI_Allreduce(AUTOPAS_MPI_IN_PLACE, &numberOfParticles, 1, AUTOPAS_MPI_UNSIGNED_INT, AUTOPAS_MPI_SUM,
-                                 AUTOPAS_MPI_COMM_WORLD);
-  // approximately the resolution we want to get.
-  size_t numberOfCells = ceil(numberOfParticles / 10.);
-  std::array<double, 3> domainSizePerDimension = {};
-  for (int i = 0; i < 3; ++i) {
-    domainSizePerDimension[i] = endCorner[i] - startCorner[i];
-  }
+  using namespace autopas::utils::ArrayMath::literals;
 
-  // get cellLength which is equal in each direction, derived from the domainsize and the requested number of cells
-  const double volume = domainSizePerDimension[0] * domainSizePerDimension[1] * domainSizePerDimension[2];
-  const double cellVolume = volume / numberOfCells;
-  const double cellLength = cbrt(cellVolume);
-
-  // calculate the size of the boundary cells, which might be smaller then the other cells
-  std::array<size_t, 3> cellsPerDimension = {};
-  // size of the last cell layer per dimension. This cell might get truncated to fit in the domain.
-  std::array<double, 3> outerCellSizePerDimension = {};
-  for (int i = 0; i < 3; ++i) {
-    outerCellSizePerDimension[i] =
-        domainSizePerDimension[i] - (floor(domainSizePerDimension[i] / cellLength) * cellLength);
-    cellsPerDimension[i] = ceil(domainSizePerDimension[i] / cellLength);
-  }
-  // Actual number of cells we end up with
-  numberOfCells = cellsPerDimension[0] * cellsPerDimension[1] * cellsPerDimension[2];
+  const auto numberOfParticles = container.getNumberOfParticles();
+  const auto domainDimensions = endCorner - startCorner;
+  const auto domainVolume = domainDimensions[0] * domainDimensions[1] * domainDimensions[2];
+
+  // We scale the dimensions of the domain to bins with volumes which give approximately 10 particles per bin.
+  const auto targetNumberOfBins = std::ceil(numberOfParticles / 10.);
+  const auto targetNumberOfBinsPerDim = std::cbrt((double)targetNumberOfBins);
+  // This is probably not an integer, so we floor to get more than 10 particles per bin than too small bins
+  const auto numberOfBinsPerDim = static_cast<int>(std::floor(targetNumberOfBinsPerDim));
+  const auto numberOfBins = numberOfBinsPerDim * numberOfBinsPerDim * numberOfBinsPerDim;
+  const auto binVolume = domainVolume / (double)numberOfBins;
+  const auto binDimensions = domainDimensions / (double)numberOfBinsPerDim;
+
+  // Calculate the actual number of bins per dimension. The rounding is needed in case the division slightly
+  // underestimates the division. The choice of dimension 0 is arbitrary.
```

Comment:
What do you mean with this? It's needed to correct floating point errors from the division right?

Reply:
Not needed anymore